src/couch_prometheus/src/couch_prometheus_util.erl

```diff
@@ -146,9 +146,9 @@ path_to_name(Path) ->
 
 counter_metric(Path) ->
     Name = path_to_name(Path),
-    case string:find(Name, <<"_total">>, trailing) == <<"_total">> of
-        true -> Name;
-        false -> to_bin(io_lib:format("~s_total", [Name]))
+    case string:find(Name, <<"_total">>, trailing) of
+        nomatch -> to_bin(io_lib:format("~s_total", [Name]));
+        _ -> Name
```

Comment:
Do we care about a suffix only? Otherwise `string:find/3` will find non-suffix `_total` strings as well.

```erlang
string:find("my_total_metric", <<"_total">>, trailing).
"_total_metric"
```

The issue previously was likely the list vs binary confusion. `string` function will return either one depending on input. To avoid any confusion a simple `lists:suffix/2` might work better?

```erlang
> true = lists:suffix("_total", "my_metric_total").
```

Reply:
thanks - the solution with `lists:suffix` looks correct as we only want to match a strict suffix. I've pushed an update which uses that.